### ISSUE TITLE
separate reward address in proof-of-stake bonding method

### ISSUE THREAD

**Issue by dckc** [rchain/rchain#3635]
**update**: see https://github.com/rchain/rchain/issues/3755

In https://github.com/rchain/rchip-proposals/issues/32#issuecomment-1065842958 @tgrospic writes:
> @dckc It would be very useful if you can create an issue with possible attack vectors if you can think of any. We discussed about your suggestion in our dev standup. @leithaus said he needs to think if there is some potential misuse but in general we had agreement that we want to implement it in PoS

### Security Considerations

After thinking about it for a few days, I don't see that this would introduce any opportunities for misuse.

I suggest that adding a separate reward address is no different from a very normal business arrangement where, for example, a performer says to the venue, "to pay me, send the check to my accountant at this address ... ."

Furthermore, this allows the reward address to be a multi-sig, which reduces risks around the rewarded funds.

Also, this is standard practice in the multi-billion dollar economy Cosmos ecosystem:

![image](https://user-images.githubusercontent.com/150986/158025507-95ffe2c6-0bc2-4b17-9683-526a51dcd40e.png)

https://www.mintscan.io/osmosis/account/osmo1hjct6q7npsspsg3dgvzk3sdf89spmlpfqua7lv

_I don't see an option for a feature request; I suggest that the **bug** label is inappropriate._

**Comment by tgrospic** [comment]
I tried to find more info on Cosmostation and it seems that Rewards Address is used by **delegators** if they want to receive rewards to different address then default.
https://guide.cosmostation.io/web_wallet_en.html
![](https://user-images.githubusercontent.com/5306205/158142937-0f23c45f-39b6-4ae6-b72b-5e966d6d0b80.png)

Validators have **Operator Address** _which token holders delegate their voting powers to._
https://www.mintscan.io/osmosis/validators/osmovaloper1hjct6q7npsspsg3dgvzk3sdf89spmlpf6t4agt
![](https://user-images.githubusercontent.com/5306205/158142831-6c9eb19e-fc93-4786-b134-a557ce7e5c21.png)

*Operator Address* seems to be what you are suggesting, but I couldn't find more information how it's used.

With this additional address users which are part of validator pool can have contract which will ensure they get their part of the rewards. What about their stake which validator collects?
Basically, I'm interested how this additional REV address can be used and is it enough to create contracts for staking pools.

**Comment by dckc** [comment]
> Rewards Address is used by **delegators**...

Ah. so what I found is not quite analagous to what I'm asking for here.

> What about their stake which validator collects?

The extra address parameter on the `bond` method is where to send the stake that the validator collects; a validator can use a staking rewards contract for that address. The users/delegators use some method on that contract to withdraw their rewards.

**Comment by tgrospic** [comment]
Thanks! Info you provided was helpful to me together with @jimscarver's input on dev standup today.

We have notion of withdrawal in PoS and the end operation of withdraw is the transfer of `stake + rewards` to the REV address of the validator's private key.
We will add _overload_ to `bond` method to optionally supply different _withdrawal_ REV address. This address can be controlled by a _staking rewards contract_ or whatever validator wants.

Is the _staking pool contact_ synonym for _staking rewards contract_?

**Comment by dckc** [comment]
> We will add overload to bond method to optionally supply different withdrawal REV address.

bingo!

> Is the staking pool contact synonym for staking rewards contract?

yes.

**Comment by dckc** [comment]
looks like this is now tracked as #3755
